### PR TITLE
Set default branchname on first push

### LINKED ISSUE #3204: Default branch is not master will display master
Only effect release v1.3 since master has another issue #3181.
image

### LINKED ISSUE #3525: Wrong label on branch dropdown
- Gitea version (or commit ref): 1.3.2
- Git version: 2.11.0
- Operating system: Debian 9
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant

https://try.gitea.io/foobar42/test-repository

## Description

When pushing a non-default branch (e.g. everything except `master`) to an empty repository, the dropdown menu for branch selection is wrongly labeled `master`. The dropdown menu shows the correct branches.

On further investigation I noticed that the link to the commits for the current branches links to https://try.gitea.io/foobar42/test-repository/commits/branch/master which produces a 404 error.

Also visiting the branches overview (https://try.gitea.io/foobar42/test-repository/branches/) lists `master` as default branch (which does not exist) but does not list the `develop` branch.

In general it seems like the UI gets confused when there is only one branch which at the same time is not the default branch.

### PR DESCRIPTION
**Pull request by Bwko**
Fixes #3204 & fixes #3525

### CODE REVIEW

File: models/action.go [hunk 1]
```diff
@@ -523,6 +523,14 @@ func CommitRepoAction(opts CommitRepoActionOptions) error {
 		return fmt.Errorf("GetRepositoryByName [owner_id: %d, name: %s]: %v", opts.RepoOwnerID, opts.RepoName, err)
 	}
 
+	refName := git.RefEndName(opts.RefFullName)
+	if repo.IsBare && refName != repo.DefaultBranch {
+		repo.DefaultBranch = refName
+		if err = repo.UpdateDefaultBranch(); err != nil {
```
Comment (lunny) [inline models/action.go]: why not update on line 536?

**Review by lunny — CHANGES_REQUESTED** [review]

**Review by jonasfranz — APPROVED** [review]

**Review by appleboy — APPROVED** [review]

**Comment by lafriks** [comment]
@lunny need your approval

**Review by lunny — APPROVED** [review]

**Comment by lafriks** [comment]
@Bwko please backport to release/v1.4 branch

**Comment by lafriks** [comment]
@Bwko small reminder ;)

**Comment by davidlt** [comment]
This is a bit annoying for people doing `git clone --mirror` / `git push --mirror` and expecting `master` to be the default. This broke my workflow because now the first pushed branch becomes defaults, which is a bit random if you do `git push --mirror` and you have to go in the settings after each `git push --mirror` to set it to `master`.

**Comment by lafriks** [comment]
@davidlt please open issue

**Pull request merged**